completers/nix-instantiate_completer/cmd/root.go

```diff
@@ -0,0 +1,61 @@
+package cmd
+
+import (
+	"github.com/rsteube/carapace"
+	"github.com/rsteube/carapace-bin/pkg/actions/tools/nix"
+	"github.com/spf13/cobra"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "nix-instantiate",
+	Short: "instantiate store derivations from Nix expression",
+	Long:  "https://nixos.org/manual/nix/stable/command-ref/nix-instantiate.html",
+	Run:   func(cmd *cobra.Command, args []string) {},
+}
+
+func Execute() error {
+	return rootCmd.Execute()
+}
+func init() {
+	carapace.Gen(rootCmd).Standalone()
+
+	arg := rootCmd.Flags().StringSlice("arg", []string{}, "Pass [name] [expr] to Nix functions.")
+	argstr := rootCmd.Flags().StringSlice("argstr", []string{}, "Pass [name] [string] to Nix functions")
+	rootCmd.Flags().StringP("attr", "A", "", "Attribute to build")
+	rootCmd.Flags().Bool("dry-run", false, "Show what store paths would be built or downloaded")
+	rootCmd.Flags().Bool("help", false, "Show usage information")
+	rootCmd.Flags().StringP("include", "I", "", "Include paths")
+	rootCmd.Flags().String("add-root", "", "Create a symlink to the output path")
+	rootCmd.Flags().StringP("expr", "E", "", "The expression to evaluate")
+	rootCmd.Flags().Bool("parse", false, "Just parse the input files, and print their abstract syntax trees on standard output in ATerm format.")
+	rootCmd.Flags().Bool("eval", false, "Evaluate the input files, and print the resulting values on standard output")
+	rootCmd.Flags().Bool("strict", false, "Recursively evaluate list elements and attributes.")
+	rootCmd.Flags().Bool("json", false, "Print the result of --eval as JSON")
+	rootCmd.Flags().Bool("xml", false, "Print the result of --eval as XML")
+	rootCmd.Flags().Bool("read-write-mode", false, "Perform evaluation in read/write mode")
```

Comment:
I think these flags only make sense when `--eval` is set is there some way to specify that?

Reply:
Something similar to this should work. But should be fine to add them by default as well.
```go
	carapace.Gen(rootCmd).PreRun(func (cmd *cobra.Command, args []string) {
		f := pflag.NewFlagSet("evalcheck", pflag.ContinueOnError)
		f.Bool("eval", false, "Evaluate the input files, and print the resulting values on standard output")
		f.Parse(args)
		if f.Lookup("eval").Changed {
			// add flags to root cmd			
		}
	})
```

Reply:
Alright, I'll just leave them there by default